packages/amplify-cli-core/src/types.ts

```diff
@@ -10,6 +10,39 @@ import { Tag } from './tags';
  */
 export type $TSAny = any; // eslint-disable-line  @typescript-eslint/no-explicit-any
 
+export type PinpointInfo = {
```

Comment:
Instead of using `$TSAny`, I think `Record<string, any>` would provide a tad more insight into these various types.

There's also a few specific types like `$TSMeta` for `amplifyMeta` and `$TSTeamProviderInfo` for `teamProviderInfo`. Those types are essentially aliases of `$TSAny`, but I think it would be good to change those to `Record<string, any>` too.

Reply:
These are things that Spencer and I are considering moving forward. The next PR will address non-primitive types and do more re-typing where I will verify whether $TSMeta, $TSTeamProviderInfo, etc are applicable to these properties. The reason I want to do this in multiple PRs is to keep things moving relatively quickly, do due diligence since we don't actually know what's inside exeInfo (it may be $TSMeta, but it may be a $TSMeta with other fields added as well) and, finally, to hopefully have real types instead of $TSMeta -> $TSAny.

Record<string, any> is something I could do here - but it wasn't clear to met yet what the benefits would be.

Reply:
I suggested it because `Record<string, any>` would narrow the type definition to objects and prevent accidentally assigning something like a boolean to one of the fields currently marked as $TSAny. Not a blocker for this PR though.

Reply:
I see, that makes sense.

Reply:
Oh! Yeah, that makes sense. But I hope to have solid types for objects as I go. I will do this for other types in my next PR. Thanks!